examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,534 @@
+module ProjPlaneCurveModule
+
+using Oscar, Markdown
+
+include("AffinePlaneCurve.jl")
+using .AffinePlaneCurveModule
+
+import Base.:(==)
+
+export ProjPlaneCurve, degree, jacobi_ideal, issmooth, tangent,
+       curve_components, isirreducible, isreduced, common_component, reduction,
+       dehomogenization, curve_intersect, curve_singular_locus, issmooth_curve
+
+################################################################################
+#
+# We follow the lecture notes by Andreas Gathmann:
+# https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+################################################################################
+#
+
+mutable struct ProjPlaneCurve{S}
+  eq::Oscar.MPolyElem_dec{S}              # Equation of the curve (polynomial in three variables)
+  degree::Int64                           # degree of the equation of the curve
+  dimension::Int64                        # dimension of the curve (as a variety)
+  components::Dict
```

Comment:
The exact `Dict` type seems to be known?
```suggestion
  components::Dict{ProjPlaneCurve{S}, Int64}
```